packages/mobile-signing-provider/esbuild.config.js

```diff
@@ -1,11 +1,467 @@
 import { baseConfig } from '@configs/esbuild';
 import * as esbuild from 'esbuild';
 import fs from 'fs';
+import path from 'path';
+import { fileURLToPath } from 'url';
 
 esbuild
   .build({
     ...baseConfig,
     entryPoints: ['./src/mobile-signing-provider.ts'],
+    // These are workarounds to avoid not needed code being bundled
+    // Always review when updating the walletconnect packages
+    plugins: [
+      {
+        name: 'bn-resolver',
+        setup(build) {
+          build.onResolve({ filter: /^bn\.js$/ }, () => {
+            return { path: 'virtual:bn.js', namespace: 'bn-shim' };
+          });
+          build.onLoad(
+            { filter: /^virtual:bn\.js$/, namespace: 'bn-shim' },
+            () => ({
+              contents: `
+                export class BN {
+                  constructor(value) {
+                    this.value = BigInt(value);
+                  }
+                  
+                  toString(base = 10) {
+                    return this.value.toString(base);
+                  }
+                  
+                  toNumber() {
+                    return Number(this.value);
+                  }
+                }
+                export default BN;
+              `,
+            })
+          );
+        },
+      },
+      {
+        name: 'customize-walletconnect',
```

Comment:
this one, probably bad idea, but adds a lot to the bundle, probably  almost not used